### PR TITLE
Fix IPZ Parser

### PR DESCRIPTION
**Pull request by RekhaAparna01**
Fixed the following issues:
-vtocECCcheck() API is failing
 1. Offset::VTOC_PTR is advanced twice in this API
 2. vtocECCcheck is checked for passing condition instead of failure 
    to throw an error

-readPT() API is reading record name wrong
  Record name is read after advancing the pointer to record offset

-readKeyword() API is appending keyword and value pair twice to the map

Test:
Tested code by parsing an ipz vpd file using vpd-parser exe

UTIL :
VSBP : DR - VPD ROOT NODE
VMPU :
MER0 :
VER0 :
VSYS : DR - SYSTEM
VEIR :
VCEN : DR - CEC
LXR0 :
VINI : HW - 0001 CC - 2E2D SN - Y131UF07302T PN - 02WG678 DR - SYSTEM BACKPLANE

### CODE REVIEW

File: src/ipz_parser.cpp [hunk 1]
```diff
@@ -312,6 +312,8 @@ types::RecordOffsetList
     auto end = itrToPT;
     std::advance(end, ptLength);
 
+    std::string recordName(itrToPT, itrToPT + Length::RECORD_NAME);
```
[the remark this answers is not in the document]
Comment (RekhaAparna01) [reply src/ipz_parser.cpp]: done

File: src/ipz_parser.cpp [hunk 2]
```diff
@@ -519,11 +517,11 @@ types::IPZVpdMap::mapped_type
                                               itrToKwds);
             kwdValueMap.emplace(std::move(kwdName), std::move(kwdValue));
         }
-
-        // support all the Keywords
-        auto stop = std::next(itrToKwds, kwdDataLength);
-        std::string kwdata(itrToKwds, stop);
-        kwdValueMap.emplace(std::move(kwdName), std::move(kwdata));
+        else
```
Comment (PriyangaRamasamy) [inline src/ipz_parser.cpp:520]: why there is no support for other keywords which are not present in the supportedKeywords list?

why parse method parses and populates only specific keywords and not all the keywords on dbus?

Comment (RekhaAparna01) [reply src/ipz_parser.cpp:520]: To handle all the keyword values for IPZ parser an issue is created - #221 . This will be taken care in that issue.

**Review by branupama — APPROVED** [review]

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]
Please fix the grammar in commit heading and body.
*  Fix the following issues: -> Fixed following issues
* Also give a brief one line about the issue that was fixed under each heading.
eg: -vtocECCcheck() API failure
Explain what was the failure on one or two lines.

File: src/ipz_parser.cpp (hunk 2 above)
Comment (PriyangaRamasamy) [reply src/ipz_parser.cpp:520]: ok

**Comment by RekhaAparna01** [comment]
> Please fix the grammar in commit heading and body.
> 
> * Fix the following issues: -> Fixed following issues
> * Also give a brief one line about the issue that was fixed under each heading.
>   eg: -vtocECCcheck() API failure
>   Explain what was the failure on one or two lines.

Done

**Review by PriyangaRamasamy — APPROVED** [review]

**Review by SunnySrivastava1984 — APPROVED** [review]

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**